src/sagemaker_xgboost_container/checkpointing.py

```diff
@@ -42,72 +42,73 @@ def train(train_args, checkpoint_dir):
 
     xgb_model, start_iteration = load_checkpoint(checkpoint_dir)
 
+    # xgboost's default value for num_boost_round is 10.
+    # If num_boost_round <= 0, xgb.train() doesn't actually train and
+    # immediately returns a Booster object.
+    train_args["num_boost_round"] = train_args.get("num_boost_round", 10) - start_iteration
+
     if xgb_model is not None:
-        logging.info("Checkpoint loaded from %s", xgb_model)
-        logging.info("Resuming from iteration %s", start_iteration)
+        logging.warning("Checkpoint loaded from %s", xgb_model)
```

Comment:
Nitpick - Why's this a warning?

Reply:
thanks for pointing it out, that should be logging.info only. I remember I was facing some assert error as caplog was not capturing the INFO level logs and hence I was just trying different logging levels and forgot to look into it later on. But I think I got the problem just not sure why it was working previously maybe something might have changed with pytest, but according to the existing pytest documentation it captures log messages of level WARNING or above, and setting the caplog level to logging.INFO explicitly prior to running this particular test fixed the problem. This would be changed in next revision.